docs/userguide/campaigns.md

```diff
@@ -1,25 +1,125 @@
+involving tasting and assigning a rating on a scale of 1 to 10, where a higher score
+indicates better quality.
+
+```python
+from baybe.targets import NumericalTarget
+from baybe.objective import Objective
+
+target = NumericalTarget(name="quality", mode="MAX", bounds=(1,10))
+objective = Objective(mode="SINGLE", targets=[target])
+```
+
+Combining the search space and the objective now enables us to create a campaign.
+
+```python
+from baybe.campaign import Campaign
+
+campaign = Campaign(searchspace=space, objective=objective)
+```
+
+## Getting a recommendation and adding a measurement
+
+To obtain a recommendation for the next experiment, we can query the campaign and use
+the [`recommend`](baybe.campaign.Campaign.recommend) function. The function takes only
+one argument, which is the `batch_quantity` keyword. This specifies the desired size of
+the batch of experiments to be conducted.
+
+```python
+rec = campaign.recommend(batch_quantity=3)
+```
+
+The `recommend` function returns a `DataFrame` with `batch_quantity` many rows, each 
+representing a set of parameters from the search space.
+To add measurements, we expand the `DataFrame` by adding a new column for the target.
+We can then provide the campaign with these measurements and receive a new
+recommendation.
+
+```python
+rec["quality"] = [2,4,9]
+campaign.add_measurements(rec)
+new_rec = campaign.recommend(batch_quantity=5)
+```
+
+## Further specifictation of campaign
+
+Although only a search space and an objective are necessary to create a campaign,
+several other aspects can be changed by the user.
+1. **A strategy:**: By default, campaigns use the composite
+[`TwoPhaseStrategy`](baybe.strategies.composite.TwoPhaseStrategy). This strategy uses a
+[`RandomRecommender`](baybe.recommenders.sampling.RandomRecommender) for the initial
+recommendation and then switches to a [`SequentialGreedyRecommender`](baybe.recommenders.bayesian.SequentialGreedyRecommender).
+However, using the `strategy` keyword, you can specify a different strategy.
+For more details on strategies, see the corresponding [`submodule`](baybe.strategies) or
+[user guide](./strategy).
+* **Previously conducted experiments**: If you have previously conducted experiments and
+have a `DataFrame` representing them, you can provide it using the `measurements_exp`
+keyword. Preferably, the `DataFrame` was created by a previous
+campaign using [the campaigns recommend function](baybe.campaign.Campaign.recommend).
+If no `DataFrame` is provided, it is assumed that no experiments were conducted previously.
+* **Numerical tolerance**: By default, numerical measurements are required to fall
+predefined tolerance. However, you can disable this requirement by uing the
+`numrical_measurements_must_be_within_tolerance` flag.
+Note that the setting of the tolerances is controlled as a part of the respective
+parameter
+* **Previously done batches and fits** If the campaign builds upon previously
```

Comment:
these are internal members and it should not be recommended to the user to touch them

Reply:
Then why are the attributes "public", i.e., do not have an underscore and are even documented?

Reply:
I have removed them. Would still be interested in why they actually exist in this way and how to handle those (see my longer comment).

Reply:
ye sthey should prob eb private, there are two reasons why theyre not i) historically and due to attrs making it a bit more awkward and ii) in some case (even though the var should never be used to `set` stuff) it can be used to `get` stuff. For instance `measurements_exp` can be useful if you simply want to retrieve all measurements you've ever added. In C++ I would just make them private and have a `get_measurements_exp` function, perhaps we need to do something similar here. But for the user guide its important that you dont encourage suers to modify these sort-of-private attributes

Reply:
I think that we could also solve these problems (at least in the short term) by adjusting the docstrings of properties to make clear how things should (not) be used.

Reply:
Sorry for the late answer but: I don't agree here. See my other comment, there I explained the actual stuff why these things are exposed via the constructor. To both of Martin's points:
1) there is no issue with attrs here, we have full control of what we want to expose and with what name. At the moment, this is done on purpose, for the reasons explained
2) Note that "being exposed via constructor" and "being public" are two different things! Ignoring the serialization issue for a moment, we could very easily remove `measurements_exp` from the constructor, but still have it as a public field. But even in this scenario, the actual question would rather be if it should be writable or only readable. So if serialization was not a problem, IMO the "correct" way would be to remove it from the constructor and have it only as a readable attribute. That would be the best setting. 

But let me think about how to solve the root cause first (i.e. serialization). Still, if you think you have a good idea, let me know!!!

Reply:
If I understood correctly, this discussion was resolved for the moment in our dev meeting, so feel free to resolve.

Reply:
Jep, will resolve